pkg/utils/set/threadsafe.go

```diff
@@ -0,0 +1,88 @@
+package set
+
+import (
+	"sync"
+)
+
+type ThreadSafeSet[T, H comparable] struct {
```

Comment:
Yep, this is good and should the the "default" Set. I would even call it Set and the other something else, but that is not important.

Reply:
In hashicorp's LRU they call the non thread safe version (wrapped by the threadsafe one) `SimpleLRU` so we can do the same semantics `SimpleSet` and `Set`.